src/Dropdown.js

```diff
@@ -4,6 +4,48 @@ import classNames from 'classnames';
 
 const DEFAULT_PLACEHOLDER_STRING = 'Select...';
 
+export const isValidLabelOrValue = value => (
+  /string|boolean|number/.test(typeof value));
+
+export const getOptionLabel = (option, label = option) => {
+  if (isValidLabelOrValue(option.label))
+    label = option.label;
```

Comment:
probably not a good idea to mutate the function params

a better way to write your function can be

```js
const getOptionLabel = (option, label = option) => {
  if (isValidLabelOrValue(option.label)) {
     return option.label;
  } 
  if (isValidLabelOrValue(option.value)){
    return option.value;
  }
  return label;
}
```


notice i didn't use `else`; this is for readability purposes. 
The else clause wasnt entirely necessary due to needing another `if`

Also notice i return the value as soon as possible,
 in this case, its a function style that prevents the mutation of the params

Reply:
this function mutates the params, but not the values sent into the function

example,
```javascript
function test (param) {
  param = 3;
}

var number = 5;
test(number);

var object = { val: 'val' };
test(object);

console.log(number); // 5
console.log(object); // val
```

If you still recommend the change, I will change the functions. If you think they are OK as-is I'm also fine with them as they are. I will make the other changes here per your suggestions.

Reply:
yup i would still recommend the change. otherwise its a bit confusing about what is happening to the values.

the new eslint rules will prevent this is well

Reply:
the functions are updated :)